src/ProgressOnderwijsUtils/SchemaReflection/CheckConstraint.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Linq;
+using Microsoft.Data.SqlClient;
+using static ProgressOnderwijsUtils.SafeSql;
+
+namespace ProgressOnderwijsUtils.SchemaReflection
+{
+    public sealed partial class DatabaseDescription
+    {
+        public struct CheckConstraintEntry : IWrittenImplicitly
+        {
+            public DbObjectId CheckConstraintObjectId { get; set; }
+            public string Name { get; set; }
+            public string Definition { get; set; }
+            public DbObjectId TableObjectId { get; set; }
+        }
+
+        public sealed class CheckConstraint
+        {
+            readonly DatabaseDescription db;
+            public readonly DbObjectId CheckConstraintObjectId;
+            public readonly string Name;
+            public readonly string Definition;
+            readonly DbObjectId tableId;
+
+            public CheckConstraint(CheckConstraintEntry entry, DatabaseDescription db)
+            {
+                CheckConstraintObjectId = entry.CheckConstraintObjectId;
+                Name = entry.Name;
+                Definition = entry.Definition;
+                tableId = entry.TableObjectId;
+                this.db = db;
+            }
+
+            public Table Table
+                => db.TryGetTableById(tableId).AssertNotNull();

```

Comment:
Dit kan je toch ook eenvoudig weg via de db-description zelf doen? Dus is echt niet nodig lijktme.

Reply:
ja dus gewoon DbObjectId tableId public maken bedoel je?

Reply:
Ja.

Reply:
De rest van de api werkt wel zo; en het voordeel daarvan is dat je dingen kan doen als Table.Columns ipv rechtstreeks de dictionaries aan elkaar knopen (en idem voor FKs e.d.)

Ik denk dat het consistenter zou zou om een user-friendly-wrapper om de constraints heen te hebben, en evt. die ook enumereerbaar te maken vanuit de Table.

Reply:
@EamonNerbonne ik weet niet goed wat je nu bedoeld met een wrapper om constraints. Om constraint ook vanuit table beschikbaar te maken vind ik ook wel mooi, maar moet TableId nu dan ook weel omgezet worden in Table?

Reply:
Als je kijkt naar de API van Table zie je wat ik bedoel - je hebt direct toegang tot gerelateerde objecten, en niet alleen ids van de objecten, zodat je niet eerst terug naar de database description hoeft te gaan om de ids te interpreteren.

Dus ja, ik zou bij de wrapper idd de Table opslaan, en niet de TableId.  De wrapper zou bijv. 1  field voor de `CheckConstraintEntry` kunnen hebben (wat een struct kan zijn), en 1 `Table`.  En die wrapper kan dan weer door de `Table`  worden opgeleverd net als Columns.